### PR TITLE
qol: purge goerli

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Removes support for goerli (per [EF's announcement](https://blog.ethereum.org/2023/11/30/goerli-lts-update)). Goerli will no longer have any hard forks, so users that do need it for some reason can use the v1 release.

### CODE REVIEW

**Comment by rkrasiuk** [comment]
test failure ref https://github.com/paradigmxyz/reth/pull/9318

File: crates/evm/src/lib.rs [hunk 1]
```diff
@@ -137,26 +135,15 @@ pub trait ConfigureEvmEnv: Send + Sync + Unpin + Clone + 'static {
...
 
     /// Fill block environment with coinbase.
-    fn fill_block_env_with_coinbase(
-        block_env: &mut BlockEnv,
-        header: &Header,
-        after_merge: bool,
-        coinbase: Address,
-    ) {
+    /// TODO: fix
+    fn fill_block_env_inner(block_env: &mut BlockEnv, header: &Header, after_merge: bool) {
```
Comment (rkrasiuk) [inline crates/evm/src/lib.rs:144]: @mattsse better name for this?

Comment (mattsse) [reply crates/evm/src/lib.rs:144]: remove it and reuse the existing `fn fill_block_env`

File: crates/primitives/Cargo.toml [hunk 2]
```diff
@@ -18,8 +18,8 @@ reth-codecs.workspace = true
 reth-ethereum-forks.workspace = true
 reth-static-file-types.workspace = true
 reth-trie-common.workspace = true
-reth-chainspec.workspace = true
 revm-primitives = { workspace = true, features = ["serde"] }
+reth-chainspec = { workspace = true, optional = true }
```
Comment (mattsse) [inline crates/primitives/Cargo.toml:22]: why is this now optional?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/primitives/Cargo.toml (hunk 2 above)
Comment (rkrasiuk) [reply crates/primitives/Cargo.toml:22]: it was used in `block_coinbase` fn, but now only for optimism feature

File: crates/evm/src/lib.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/evm/src/lib.rs:144]: https://github.com/paradigmxyz/reth/pull/9322

**Review by mattsse — APPROVED** [review]
rip goerli

**Pull request merged**